cardano-api/src/Cardano/Api/Block.hs

```diff
@@ -251,6 +254,13 @@ data ChainPoint = ChainPointAtGenesis
                 | ChainPoint !SlotNo !(Hash BlockHeader)
   deriving (Eq, Show)
 
+instance ToJSON ChainPoint where
+  toJSON ChainPointAtGenesis = Aeson.String "Tip is currently at genesis block"
```

Comment:
This is a bit stringly typed. How about a JSON null?

A JSON instance should be by default for machines, and in a API like this we have to go with sane defaults. We cannot tailor it to the presentation needs of one specific application.